monkey/monkey_island/cc/ui/src/components/attack/techniques/Helpers.js

```diff
@@ -15,7 +15,7 @@ export function renderMachineFromSystemData(data) {
   }
   data['ips'].forEach(function (ipInfo) {
     if (typeof ipInfo === 'object') {
-      machineStr += ipInfo['addr'] + ', ';
+      machineStr += ipInfo.join(', ') + ', ';
```

Comment:
This logic is going to cause other bugs. Mainly: 
![image](https://user-images.githubusercontent.com/19957806/127861439-3b197d0b-9363-40e5-9e05-1f51f7977a03.png)

From what I can tell, `ipInfo` can be one of 3 types:
1. Object
2. Array
3. String

It just so happens that in Javascript, an array is an object, and that's why this fix works. But in the event that `ipInfo` is an object that is not an array, it causes the above error.

The solution for this is for us to make everything that produces `ipInfo` produce the same type of data. If that isn't feasible, we need to add another if to check if the type is array. Note that the array check needs to come before the object check, otherwise the array check will never be reached.

We should very strongly prefer the first solution.

Reply:
Created #1394.